// Copyright (c) SandboxAQ. All rights reserved.
// SPDX-License-Identifier: AGPL-3.0-only

// Public enums used by the I/O interface from the Sandwich C++ library.
syntax = "proto3";

package saq.sandwich.proto.io;

option go_package = "github.com/sandbox-quantum/sandwich/go/proto/sandwich";

// Error code, throwable by an I/O interface.
enum IOError {
  // No error.
  IOERROR_OK = 0;

  // Not ready yet.
  //
  // This code is returned when the connection is in progress, i.e. write is
  // not possible yet.
  IOERROR_IN_PROGRESS = 1;

  // The underlying I/O interface is non blocking and the requested operation
  // would block.
  IOERROR_WOULD_BLOCK = 2;

  // The connection was refused.
  IOERROR_REFUSED = 3;

  // Closed pipe.
  IOERROR_CLOSED = 4;

  // The underlying I/O interface is invalid.
  IOERROR_INVALID = 5;

  // Unknown I/O error.
  IOERROR_UNKNOWN = 6;

  // Next id: 7
};
